internal/server/kubernetes/client.go

```diff
@@ -449,6 +449,18 @@ func (c *client) Patch(
 	return client.Patch(ctx, obj, patch, opts...)
 }
 
+func (c *client) Apply(
+	ctx context.Context,
+	obj runtime.ApplyConfiguration,
+	opts ...libClient.ApplyOption,
+) error {
+	// For Apply operations, we delegate directly to the internal client
+	// since ApplyConfiguration doesn't provide easy access to metadata
+	// for authorization checks. This is a limitation that may need to be
+	// addressed in the future.
+	return c.internalClient.Apply(ctx, obj, opts...)
```

Comment:
This feels dangerous. If we got by without Apply up until now, please just make this return an error instead and we can fix _that_ if/when needed.